ludwig/utils/fs_utils.py

```diff
@@ -53,6 +57,46 @@ def upgrade_http(urlpath):
     return None
 
 
+def get_bytes_obj_if_path(path: Any) -> Union[Any, Optional[bytes]]:
+    """Gets bytes string if `path` is a path (e.g. a string).
+
+    If it is not a path, return as-is.
+    """
+    if not isinstance(path, str):
+        return path
+    return get_bytes_obj_from_path(path)
+
+
+@functools.lru_cache(maxsize=32)
```

Comment:
Does this get called repeatedly for the same inputs?

Reply:
The `lru_cache` decorator was present in the original `audio_utils.read_audio` function, so I just preserved that functionality here. Might be useful if there are duplicate paths in the dataset, but happy to remove it if you think it is unnecessary.

Reply:
We can leave it in for now and revisit down the road, I suppose.